wave_kinematics/src/rotation.cpp

```diff
@@ -0,0 +1,99 @@
+#include "wave/kinematics/rotation.hpp"
+#include "wave/utils/utils.hpp"
+
+double wave::Rotation::comparision_threshold = 1e-8;
+
+wave::Rotation::Rotation() {}
+
+void wave::Rotation::setToIdentity() {
+  this->rotation_object.setIdentity();
+}
+
+void wave::Rotation::compose(const Rotation &R) {
+  this->rotation_object = this->rotation_object * R.rotation_object;
+}
+
+wave::Mat3 wave::Rotation::invert() {
+  this->rotation_object.invert();
+}
+
+wave::Vec3 wave::Rotation::rotate(const Vec3 &input_vector) {
+  return this->rotation_object.rotate(input_vector);
+}
+
+wave::Vec3 wave::Rotation::inverseRotate(const Vec3 &input_vector) {
+  return this->rotation_object.inverseRotate(input_vector);
+}
+
+wave::Vec3 wave::Rotation::rotateAndJacobian(const Vec3 &input_vector,
```

Comment:
ahh ok, so the suggestion is to decouple the function and jacobian.  I disagree that is clearer for this use case.  The Jacobian is always specific to the input and output relationship to the function.  For example, if there is a function F that takes in 5 different parameters (not necessarily vector elements, could be manifold quantities):
``` y = F(x1,x2,x3,x4,x5) ```
I think it's cleaner to have the function
``` FandJacobian(x1,x2,x3,x4,x5, J_x1, J_x2, J_x3 ...)```
Which is pretty clear in saying that J_x1 is the jacobian of the underlying function F, wrt the x1 parameter, etc, rather than
```  y = F(x1,x2,x3,x4,x5 ) 
    dy_x1 = FandJacobianX1(x1,x2,x3,x4,x5)
    dy_x2 = FandJacobianX2(x1,x2,x3,x4,x5)
  ...
```
Which feels like unnecessary bloat to me, and sometimes will be duplicating calculations since we will loose intermediate calculations